std/typecons.d

```diff
@@ -768,6 +770,12 @@ if (distinctFieldNames!(Specs))
             return field[] == rhs.field[];
         }
 
+        bool opEquals(R...)(R rhs) const
```

Comment:
auto ref may be helpful here

Reply:
Yeah, but then DMD can no longer find a matching `opEquals`

Reply:
Can you file a bug for `auto ref opEquals`?

Reply:
(with the latest constraint + body changes, I can no longer reproduce this -> changed to `const auto ref`)